src/components/app/data/hooks/useAlgoliaSearch.ts

```diff
@@ -0,0 +1,181 @@
+import { getConfig } from '@edx/frontend-platform';
+import { useEffect, useMemo } from 'react';
+import algoliasearch from 'algoliasearch';
+import { logError } from '@edx/frontend-platform/logging';
+import useBFF from './useBFF';
+import useEnterpriseFeatures from './useEnterpriseFeatures';
+import useEnterpriseCustomer from './useEnterpriseCustomer';
+import { queryDefaultEmptyFallback } from '../queries';
+
+type ExtractAlgoliaArgs = {
+  data: SecuredAlgoliaApiData | null;
+  isCatalogQueryFiltersEnabled: boolean;
+  isIndexSupported: boolean;
+};
+
+/**
+ * Extracts secured Algolia metadata from backend data based on feature and index support flags.
+ *
+ * Returns the actual secured Algolia API key and catalog UUID mapping only if:
+ * - catalog query filters are enabled
+ * - the given index is supported
+ * - valid data is provided
+ *
+ * Otherwise, returns a default object with `null` API key and empty mapping.
+ *
+ * @param data - Backend response data containing Algolia metadata.
+ * @param isCatalogQueryFiltersEnabled - Whether catalog query filters are enabled for the enterprise customer.
+ * @param isIndexSupported - Whether the specified Algolia index is supported for secured search.
+ *
+ * @returns An object with the secured Algolia API key and the catalog UUID to query UUID mapping.
+ */
+const extractAlgolia = ({
+  data,
+  isCatalogQueryFiltersEnabled,
+  isIndexSupported,
+}: ExtractAlgoliaArgs) => (
+  isCatalogQueryFiltersEnabled && isIndexSupported && data
```

Comment:
nit: i feel using regular `if` conditional vs. ternary here would be slightly more readable

```tsx
if (isCatalogQueryFiltersEnabled && isIndexSupported && data) {
  return {
    securedAlgoliaApiKey: data.securedAlgoliaApiKey,
    catalogUuidsToCatalogQueryUuids: data.catalogUuidsToCatalogQueryUuids,
  };
}
return {
  securedAlgoliaApiKey: null,
  catalogUuidsToCatalogQueryUuids: {},
};
```